build.gradle

```diff
@@ -1,208 +1,36 @@
-    def snippet = ""
-    def stop = false
-    changelog.eachLine { line, count ->
-        if (count >= 2) {
-            stop = stop || line.startsWith("v");
-            if (!stop) {
-                snippet += line + "\n";
-            }
-        }
-    }
-    return "  " + snippet.trim();
-}
-
-def checkAndDefaultProperty(prop) {
-    if (!project.hasProperty(prop)) {
-        logger.warn("Add " + prop + " to your ~/.gradle/gradle.properties file.")
-        rootProject.ext.set(prop, prop)
-    }
-}
-
-def checkForChangelogUpdates(task) {
-    def changelogtext = rootProject.file('CHANGELOG.md').text
-    if (!changelogtext.startsWith("v${unsnapshottedVersion} -")) {
-        throw new AssertionError(
-                "Changelog must be updated with v{$unsnapshottedVersion} before release. Please check " +
-                        rootProject.file('CHANGELOG.md').absolutePath)
-    }
-}
-
-gradle.taskGraph.afterTask { Task task, TaskState state ->
-    if (task.path.endsWith("release") || task.path.endsWith("githubReleaseZip")
-            || task.path.endsWith("publicrepoDistZip")) {
-        checkForChangelogUpdates(task)
-    }
-}
-
-// Skip signing archives on Jenkins when -SNAPSHOT is being checked in.
-gradle.taskGraph.beforeTask { Task task ->
-    if (task.path.contains("sign") && !ext.isReleaseVersion) {
-        task.enabled = false
-    }
-}
-
-task updateChangelog() << {
-    def newVersion = findProperty("version").replaceAll('-SNAPSHOT', '')
-    def changelog = rootProject.file('CHANGELOG.md')
-    def changelogText = changelog.text
-    if (!changelogText.startsWith("v${newVersion} -")) {
-        def updatedChangelog = "v${newVersion} - TBD\n"
-        def dashesCount = updatedChangelog.length()-1
-        updatedChangelog += "-"*dashesCount + "\n\n"
-
-        changelog.write(updatedChangelog + changelogText)
+    dependencies {
+        classpath deps.build.gradlePlugins.github
+        classpath deps.build.gradlePlugins.release
     }
 }
 
-afterReleaseBuild.dependsOn(":core-android:uploadArchives", ":rides-android:uploadArchives")
-updateVersion.dependsOn ":githubRelease"
-commitNewVersion.dependsOn ':updateChangelog'
-githubRelease.dependsOn project(":samples").subprojects.collect { it.path + ":githubReleaseZip" }
-
-release {
-    failOnCommitNeeded = false
-    failOnPublishNeeded = false
-    failOnSnapshotDependencies = false
-    revertOnFail = true
-    tagTemplate = "v${unsnapshottedVersion}"
-}
-
-github {
-    owner = 'uber'
-    repo = 'rides-android-sdk'
-    token = "${githubToken}"
-    tagName = "v${unsnapshottedVersion}"
-    targetCommitish = 'master'
-    name = "v${unsnapshottedVersion}"
-    body = generateReleaseNotes()
-    assets = project.samples.collect {
-        project(it).buildDir.absolutePath + "/distributions/" + project(it).name +
-                "-v${unsnapshottedVersion}.zip"
-    }
+task wrapper(type: Wrapper) {
+    gradleVersion = '4.2.1'
```

Comment:
nit: 4.3 is out, with better classloader support for working with multiple modules